puppet/modules/redmine/manifests/init.pp

```diff
@@ -161,10 +162,59 @@
   $docroot          = "${app_root}/public"
   $priority         = '05'
 
-  $apache_backend_config = {
-    passenger_app_root      => $app_root,
-    passenger_min_instances => 1,
-    passenger_start_timeout => 600,
+  if $deployment == 'passenger' {
+    $apache_backend_config = {
+      passenger_app_root      => $app_root,
+      passenger_min_instances => 1,
+      passenger_start_timeout => 600,
+    }
+  } else {
+    systemd::manage_unit {'redmine.socket':
+      ensure        => 'present',
+      unit_entry    => {
+        'Description' => 'redmine socket',
+      },
+      socket_entry  => {
+        'ListenStream' => 3000,
+      },
+      install_entry => {
+        'WantedBy' => 'sockets.target',
+      },
+    }
+
+    systemd::manage_unit{'redmine.service':
+      ensure        => 'present',
+      enable        => true,
+      active        => true,
+      unit_entry    => {
+        'Description'   => 'redmine',
+      },
+      service_entry => {
+        'Type'             => 'notify',
+        'User'             => $username,
+        #'PrivateTmp'      => true,
+        'WorkingDirectory' => $app_root,
+        'ExecStart'        => "${app_root}/bin/rails server --environment production",
+      },
+    }
+
+    $apache_backend_config = {
+      'proxy_preserve_host' => true,
+      'proxy_add_headers'   => true,
+      'request_headers'     => ['set X_FORWARDED_PROTO "https"'],
```

Comment:
In Foreman we also unset SSL client cert parameters. Is that not needed here because Redmine ignores those?